ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs

```diff
@@ -925,7 +930,10 @@ withConnectionManager ConnectionManagerArguments {
                  traceWith tracer (TrPruneConnections (Map.keysSet pruneMap)
                                                       numberToPrune
                                                       (Map.keysSet choiceMap))
-                 forM_ pruneMap $ \(_, connThread', _) -> cancel connThread'
+                 forM_ pruneMap $ \(_, connThread', _) ->
+                                   throwTo (asyncThreadId (Proxy :: Proxy m)
```

Comment:
It would be good with a comment explaining why we use `throwTo` here.

Reply:
Done, as for all the other comments.